### DISCUSSION TITLE
Active space transformations

### DISCUSSION THREAD

**Discussion by tommontgomery450** [BoothGroup/Vayesta#74] [Q&A] [answered]
Hi I have been playing around with the idea of applying additional active space transformations to the effective hamiltonian of the clusters in the DMET calculations particularly applied to molecular systems. It would be great if you were able to validate my understanding of the implementation or had any thoughts on other approaches that might work better.

My understanding is that:

After defining a fragment and then generating the bath orbitals these combine to define the cluster space. The local fragment and bath orbitals are then rotated to a canonical basis i.e. one that diagonalises the Fock matrix over the cluster space; Lets call these |i> with fock energy e_i. In this basis it seems that a check is done to make sure that the 1-RDM of the HF solution is approximately diagonal with diagonal values ~ 2 and 0 (RHF) how close depends on the threshold for DMET I am assuming) and thus the electron number is close to some integer value if this threshold is small. Once the effective hamiltonian for the cluster is defined the cluster solver will solve it to get the ground state - then uses this to get the expectation of energy over just the fragment orbitals which then feeds into the main DMET loop.

A common approximation for molecular systems is an active space transformation based on energy and occupation of molecular orbitals like the frozen core approximation or HOMO-LUMO active space transformation. Of course the accuracy of applying such transformations depends on the nature of the system but I was wondering that since a DMET cluster essentially forms acts like an effective sub-molecule with an effective hamiltonian I was wondering if it was legitimate to apply something like a HOMO LUMO / frozen core approximation based on the energy and approximate occupation of the basis |i> essentially applying an active space transformation on top of the active space transformation applied by DMET.  I think this is what they propose in https://arxiv.org/pdf/2110.08163.pdf in order to keep the fragment size large i.e. over lots of atoms but reduce the size of computation for the solver. 

Thanks for your time

Tom

**Comment by ghb24** [comment, marked as answer]
Hi Tom - thanks for the question.

If I understand your comment correctly, you are asking about doing an 'active space' method (e.g. CASSCF or CASCI) as the solver for the cluster. This would therefore mean selecting a low-energy active space set of orbitals within the cluster. There are two ways you could envisage this happening within Vayesta. Within a solely DMET scheme, you could select a large fragment, and then have to pick an active space (presumably just energetically) from within the semi-canonicalized cluster orbitals. Alternatively, you could imagine picking a small fragment, and augmenting the bath space not just with the DMET bath orbitals, but also additional orbitals (e.g. the BNOs or other options in Vayesta). Then, the 'active space' within the resulting cluster could just be the DMET and fragment semi-canonicalized orbitals within the cluster. This (in my opinion) is a more natural approach to selecting a more multi-tiered approach to the correlated treatment. We don't have this functionality natively supported in Vayesta, but we do have a TCCSD solver, which will perform the latter approach, but additionally tailor the rest of the cluster space via the T2 amplitudes of the strongly correlated region.
